ledger/src/slot_stats.rs

```diff
@@ -0,0 +1,90 @@
+use {
+    crate::blockstore_meta::SlotMeta, bitflags::bitflags, lru::LruCache, solana_sdk::clock::Slot,
+};
+
+const SLOTS_STATS_CACHE_CAPACITY: usize = 300;
+
+macro_rules! get_mut_entry (
+    ($cache:expr, $key:expr) => (
+        match $cache.get_mut(&$key) {
+            Some(entry) => entry,
+            None => {
+                $cache.put($key, SlotStats::default());
+                $cache.get_mut(&$key).unwrap()
+            }
+        }
+    );
+);
```

Comment:
any benefit of making this a macro vs implementing it as a method in `SlotsStats`?

Reply:
I actually do prefer to ditch the macro.
But I couldn't get it compile as a function, because of lifetime errors.
Feel free to give it a shot, and let me know.

Reply:
this worked for me:
```
    fn get_or_default_with_eviction_check<'a>(
        stats: &'a mut MutexGuard<LruCache<Slot, SlotStats>>,
        slot: Slot,
    ) -> (&'a mut SlotStats, Option<(Slot, SlotStats)>) {
        let mut evicted = None;
        if stats.len() == stats.cap() {
            match stats.peek_lru() {
                Some((s, _)) if *s == slot => (),
                _ => {
                    evicted = Some(stats.pop_lru().expect("cache capacity should not be 0"));
                }
            }
        }
        stats.get_or_insert(slot, SlotStats::default);
        (stats.get_mut(&slot).unwrap(), evicted)
    }
```